src/Backend/PowsyblBackend.py

```diff
@@ -0,0 +1,1540 @@
+from grid2op.Action._BackendAction import _BackendAction
+from grid2op.Action import ActionSpace
+from grid2op.Rules import RulesChecker
+from .network import load as load_ppow_network
+
+BUS_EXTENSION = '_dummy'
+
+
+try:
+    import numba
+
+    numba_ = True
+except (ImportError, ModuleNotFoundError):
+    numba_ = False
+    warnings.warn(
+        "Numba cannot be loaded. You will gain possibly massive speed if installing it by "
+        "\n\t{} -m pip install numba\n".format(sys.executable)
+    )
+class PowsyblBackend(Backend):
+
+    def __init__(
+            self,
+            detailed_infos_for_cascading_failures=False,
+            dist_slack=False,
+            can_be_copied=True,
+    ):
+        Backend.__init__(
+            self,
+            detailed_infos_for_cascading_failures=detailed_infos_for_cascading_failures,
+            dist_slack=dist_slack,
+            can_be_copied=can_be_copied
+        )
+
+        self._dist_slack = dist_slack
+
+        self.prod_pu_to_kv = None
+        self.load_pu_to_kv = None
+        self.lines_or_pu_to_kv = None
+        self.lines_ex_pu_to_kv = None
+        self.storage_pu_to_kv = None
+
+        self.p_or = None
+        self.q_or = None
+        self.v_or = None
+        self.a_or = None
+        self.p_ex = None
+        self.q_ex = None
+        self.v_ex = None
+        self.a_ex = None
+
+        self.load_p = None
+        self.load_q = None
+        self.load_v = None
+
+        self.storage_p = None
+        self.storage_q = None
+        self.storage_v = None
+
+        self.prod_p = None
+        self.prod_q = None
+        self.prod_v = None
+        self.line_status = None
+
+        self._nb_bus_before = None
+
+        self.can_output_theta = True  # I support the voltage angle
+        self.theta_or = None
+        self.theta_ex = None
+        self.load_theta = None
+        self.gen_theta = None
+        self.storage_theta = None
+
+        # self._iref_slack = None
+
+        self._topo_vect = None
+        self._init_bus_load = None
+        self._init_bus_gen = None
+        self._init_bus_lor = None
+        self._init_bus_lex = None
+        self._get_vector_inj = None
+        self._big_topo_to_obj = None
+        self._big_topo_to_backend = None
+        self.__pp_backend_initial_grid = None
+
+        # Mapping some fun to apply bus updates
+        self._type_to_bus_set = [
+            self._apply_load_bus,
+            self._apply_gen_bus,
+            self._apply_lor_bus,
+            self._apply_trafo_hv,
+            self._apply_lex_bus,
+            self._apply_trafo_lv,
+        ]
+
+        self.dim_topo = -1
+        self._number_true_line = -1
+        self.cst_1 = dt_float(1.0)
+
+        self.tol = None
+        self.map_sub = {}
+        self.__nb_bus_before = None  # number of substation in the powergrid
+        self.__nb_powerline = (
+            None  # number of powerline (real powerline, not transformer)
+        )
```

Comment:
All of these attributes are direct copy paste from the one fond in PandaPowerBackend. They are the results of profiling + optimization dedicated to pandapower. You reused them but i'm not sure pandapower and pypowsybl suffer from the same bottleneck.  Again, copy paste is not a great practice, and i'm pretty sure at least some of this code is not used in your backend.

Would have been (way) better to start, as said in the beginning of the project either with the examples https://github.com/rte-france/Grid2Op/tree/master/examples/backend_integration, the doc https://grid2op.readthedocs.io/en/latest/createbackend.html or the Educ backend made for that purpose (which is the same as the "PandapowerBackend" but without all the optimization for computation time here https://github.com/rte-france/Grid2Op/blob/master/grid2op/Backend/educPandaPowerBackend.py.

This would have also lead to a much smaller file: ~500 lines instead of more than 1500 which in turn decreases the likelihood of bugs, increase the readability and decrease the time to find the bugs if any.